lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -1654,13 +1654,22 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
           depth = rewriter.create<Torch::AtenIntScalarOp>(
               loc, rewriter.getType<Torch::IntType>(), depth);
 
-        auto selectTy = rewriter.getType<Torch::ValueTensorType>(
-            llvm::SmallVector<int64_t>{1}, valuesTy.getDtype());
-
+        // Apply bounds checking on the indices
```

Comment:
Would be easier to understand if this comment is either more concise or more specific. E.g. I personally would have written something like `// If indices < 0, indices += depth`

another perhaps better option is to remove this comment entirely - - the code is decently self-explanatory

Reply:
Removed this comment